rs/consensus/vetkd/src/lib.rs

```diff
@@ -426,3 +428,723 @@ fn reject_if_invalid(
+            assert_eq!(payload.vetkd_agreements.len(), 2);
+            for (id, context) in contexts {
+                match context.key_id() {
+                    MasterPublicKeyId::Ecdsa(_) | MasterPublicKeyId::Schnorr(_) => {
+                        assert!(!payload.vetkd_agreements.contains_key(&id));
+                    }
+                    MasterPublicKeyId::VetKd(_) => {
+                        assert_matches!(
+                            payload.vetkd_agreements.get(&id).unwrap(),
+                            VetKdAgreement::Reject(VetKdErrorCode::InvalidKey)
+                        );
+                    }
+                }
+            }
+
+            let validation =
+                builder.validate_payload(HEIGHT, &proposal_context, &serialized_payload, &[]);
+            assert!(validation.is_ok());
+
+            // payload with different rejects for the same contexts should be rejected
+            let payload = as_bytes(make_vetkd_agreements_with_payload(
+                &[1, 2],
+                VetKdAgreement::Reject(VetKdErrorCode::TimedOut),
+            ));
+            let validation = builder.validate_payload(HEIGHT, &proposal_context, &payload, &[]);
+            assert_matches!(
+                validation.unwrap_err(),
+                ValidationError::InvalidArtifact(InvalidPayloadReason::InvalidVetKdPayload(
+                    InvalidVetKdPayloadReason::MismatchedAgreement { expected, received }
+                )) if expected == Some(VetKdAgreement::Reject(VetKdErrorCode::InvalidKey))
+                   && received == Some(VetKdAgreement::Reject(VetKdErrorCode::TimedOut))
+            );
+
+            // payload with success responses for the same contexts should be rejected
+            let payload = as_bytes(make_vetkd_agreements_with_payload(
+                &[1, 2],
+                VetKdAgreement::Success(vec![1, 1, 1]),
+            ));
+            let validation = builder.validate_payload(HEIGHT, &proposal_context, &payload, &[]);
+            assert_matches!(
+                validation.unwrap_err(),
+                ValidationError::InvalidArtifact(InvalidPayloadReason::InvalidVetKdPayload(
+                    InvalidVetKdPayloadReason::MismatchedAgreement { expected, received }
+                )) if expected == Some(VetKdAgreement::Reject(VetKdErrorCode::InvalidKey))
+                   && received.is_none()
+            );
+
+            // Empty payloads should always be valid
+            let validation = builder.validate_payload(HEIGHT, &proposal_context, &[], &[]);
+            assert!(validation.is_ok());
+        })
+    }
+
+    #[test]
+    fn test_reject_timed_out_contexts() {
+        let config = make_chain_key_config();
+        let contexts = make_contexts(&config);
+        let shares = make_shares(&contexts);
+        let context = ValidationContext {
+            time: UNIX_EPOCH.checked_add(Duration::from_secs(2)).unwrap(),
```

Comment:
Tiny Nit: If you do `checked_add().unwrap()` you get the same as if you simply do an unchecked add, right?